app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -91,6 +98,37 @@ static CompletableFuture updateDocument(String key, Map<String, Object> updates,
     return getAllFuture.thenApply(querySnapshot -> querySnapshot.toObjects(cls));
   }
 
+  /**
+   * It is a temporary, simpler version to retrieve favors in a **square area** on sphere surface.
+   * This function retrieve longitude and latitude bounded documents and save them in MainActivity.
+   * For some reason, reading from db is very slow when adding one more whereEqualTo query field
+   * TODO: use firebase functions or other server code to write logic that performs customized filtering and fetch the result
+   */
+  static <T extends Document> CompletableFuture<List<T>> getAllDocumentsLongitudeLatitudeBounded(
+      Location loc, double radius, Class<T> cls, String collection, Activity activity) {
+    double longDif = Math.toDegrees(radius / (6371 * Math.cos(Math.toRadians(loc.getLatitude()))));
+    double latDif = Math.toDegrees(radius / 6371);
+    Task<QuerySnapshot> getAllTask = getCollectionReference(collection)
+            .whereGreaterThan("location.longitude", loc.getLongitude() - longDif)
+            .whereLessThan("location.longitude", loc.getLongitude() + longDif).limit(30).get();
+    CompletableFuture<QuerySnapshot> getAllFuture = new TaskToFutureAdapter<>(getAllTask).getInstance();
+    CompletableFuture<List<T>> documents = getAllFuture.thenApply(
+            querySnapshot -> querySnapshot.toObjects(cls));
+    documents.thenAccept(
```

Comment:
The "thenAccept" behaviour should be done in the mapspage rather than in the database wrapper. This will make the database wrapper method more flexible, and allow us to get for example nearby users.

Reply:
done